utils/google_drive_helper.go

```diff
@@ -1,67 +1,155 @@
 package utils
 
 import (
-    "context"
-    "io/ioutil"
+	"context"
+	"fmt"
+	googleapi "google.golang.org/api/googleapi"
+	"io/ioutil"
+	"os"
+	"regexp"
 
-    "google.golang.org/api/docs/v1"
-    "google.golang.org/api/drive/v3"
-    "google.golang.org/api/option"
 	"golang.org/x/oauth2/google"
-	
+	"google.golang.org/api/docs/v1"
+	"google.golang.org/api/drive/v3"
+	"google.golang.org/api/option"
 )
 
 type GoogleDriveHelper struct {
-    ctx    context.Context
-    srv    *drive.Service
-    docsSrv *docs.Service
+	ctx     context.Context
+	srv     *drive.Service
+	docsSrv *docs.Service
+}
+
+func GetGoogleDocIDFromURL(url string) (string, error) {
+	re := regexp.MustCompile(`\/d\/(.+?)\/`)
+	matches := re.FindStringSubmatch(url)
+	if len(matches) < 2 {
+		return "", fmt.Errorf("Could not extract google document ID from URL")
+	}
+	return matches[1], nil
+}
+
+func getGoogleIdFromInput(inputString string) (string, error) {
+	urlPattern := `^https?://.*$`
+	if matched, _ := regexp.MatchString(urlPattern, inputString); matched {
+		parserd_id, err := GetGoogleDocIDFromURL(inputString)
+		return parserd_id, err
+	}
+	return inputString, nil
+}
+
+func isGoogleDocsDocumentID(url string) bool {
+	googleDocsURLPattern := `^https?://docs.google.com/document/d/[-_a-zA-Z0-9]+/.*$`
+	regex := regexp.MustCompile(googleDocsURLPattern)
+	return regex.MatchString(url)
 }
 
 func NewGoogleDriveHelper(credentialsFilePath string) (*GoogleDriveHelper, error) {
-    ctx := context.Background()
-    data, err := ioutil.ReadFile(credentialsFilePath)
-    if err != nil {
-        return nil, err
-    }
-
-    conf, err := google.JWTConfigFromJSON(data, drive.DriveReadonlyScope, docs.DocumentsReadonlyScope)
-    if err != nil {
-        return nil, err
-    }
-
-    srv, err := drive.NewService(ctx, option.WithHTTPClient(conf.Client(ctx)))
-    if err != nil {
-        return nil, err
-    }
-
-    docsSrv, err := docs.NewService(ctx, option.WithHTTPClient(conf.Client(ctx)))
-    if err != nil {
-        return nil, err
-    }
-
-    return &GoogleDriveHelper{
-        ctx:    ctx,
-        srv:    srv,
-        docsSrv: docsSrv,
-    }, nil
+	checkPassed, err := googleHelperChecks(credentialsFilePath)
+	if err != nil {
+	}
```

Comment:
this does not look right at all, should you return err? why do you check if err is not nil?